File: Programs/Sum_of_Max_and_Min.cpp
#include<bits/stdc++.h>
using namespace std;

// CodingNinjas
// 1. Ternary Operator
int sumOfMaxMin(int arr[], int n) {
	int minimum= INT_MAX;
	int maximum= INT_MIN;

	for(int i=0; i<n; i++) {
		minimum= arr[i] < minimum? arr[i]: minimum;
		maximum= arr[i] > maximum? arr[i]: maximum;
	}
	return minimum+maximum;
}

// 2. Built-in Functions
int sumOfMaxMin(int arr[], int n) {
	int minimum= INT_MAX;
	int maximum= INT_MIN;

	for(int i=0; i<n; i++) {
		minimum= min(minimum, arr[i]);
		maximum= max(maximum, arr[i]);
	}
	return minimum+maximum;
}

File: Programs/Sum_of_Max_and_Min.py
# CodingNinjas
# 1. Use sys
from sys import maxsize
def sumOfMaxMin(arr):
    minValue, maxValue= maxsize, -maxsize-1
    for num in arr:
        minValue= num if num < minValue else minValue
        maxValue= num if num > maxValue else maxValue
    return minValue + maxValue

# 2. Use ∞
def sumOfMaxMin(arr):
    minValue, maxValue= int('inf'), int('-inf')
    for num in arr:
        minValue= num if num < minValue else minValue
        maxValue= num if num > maxValue else maxValue
    return minValue + maxValue

#  3. Use built-in Functions
def sumOfMaxMin(arr):
    return min(arr) + max(arr)